pep-0659.rst

```diff
@@ -312,32 +264,49 @@ Memory use
 
 An obvious concern with any scheme that performs any sort of caching is
 "how much more memory does it use?".
-The short answer is "none".
+The short answer is "not that much".
+
+Memory use in 3.10
+''''''''''''''''''
+
+CPython 3.10 used 2 bytes per instruction, until the execution count
+reached ~2000 when another byte per instruction and 32 bytes per
+instruction with a cache (``LOAD_GLOBAL`` and ``LOAD_ATTR``).
 
 Comparing memory use to 3.10
 ''''''''''''''''''''''''''''
+
+CPython 3.10 used 2 bytes per instruction, until the execution count
+reached ~2000 when it allocates another byte per instruction and
+32 bytes per instruction with a cache (``LOAD_GLOBAL`` and ``LOAD_ATTR``).
+
 The following table shows the additional bytes per instruction to support the
 3.10 opcache or the proposed adaptive interpreter, on a 64 bit machine.
 
-================   =====  ========  =====  =====
- Version           3.10   3.10 opt   3.11   3.11
- Specialised       20%      20%      25%    33%
-----------------   -----  --------  -----  -----
- quickened code     0        0       2      2
- opcache_map        1        1       0      0
- opcache/data       6.4     4.8      4      5.3
-----------------   -----  --------  -----  -----
- Total              7.4     5.8      6      7.3
-================   =====  ========  =====  =====
+================   ==========  ==========  ======
+ Version           3.10 cold    3.10 hot    3.11
+ Specialised           0%        ~15%       ~25%
+----------------   ----------  ----------  ------
+ code                 2           2          2
+ opcache_map          0           1          0
+ opcache/data         0          4.8         4
+----------------   ----------  ----------  ------
+ Total                2          7.8         6
+================   ==========  ==========  ======
 
-``3.10`` is the current version of 3.10 which uses 32 bytes per entry.
-``3.10 opt`` is a hypothetical improved version of 3.10 that uses 24 bytes
-per entry.
+``3.10 cold`` is before the code has reached the ~2000 limit.
+``3.10 hot`` shows the cache use once the threshold is reached.
 
-Even if one third of all instructions were specialized (a high proportion),
-then the memory use is still less than that of 3.10.
-With a more realistic 25%, then memory use is basically the same as the
-hypothetical improved version of 3.10.
+The relative memory use depends on how much code is "hot" enough to trigger
+creation of the cache in 3.10. The break even point, where the memory used
+by 3.10 is the same as for 3.11 is ~70%.
+
+It is also worth noting that the actual bytecode is only part of a code
+object. Code objects also include names, constants and quite a lot of
+debugging information.
```

Comment:
Maybe touch on the line/column-table issue when adding additional cache entries? Could be worded as "the format of these (cold) tables may need to be updated to avoid additional memory use" or something.

Reply:
I'm not sure that is relevant. We would probably want to revisit how we store the debug information even if we kept the original bytecode format.